tests/testMergeStructs.m

```diff
@@ -0,0 +1,21 @@
+classdef testMergeStructs < matlab.unittest.TestCase   
+
+    properties (TestParameter)
+        s = {struct('s1', 1, 's2', 2),...
+             struct('s1', 1),...
+             struct('s1', 1, 's2', 2, 's3', 3)};
+        d = {struct('d1', 1, 'd2', 2),...
+             struct('d1', 1),...
+             struct('d1', 1, 'd2', 2)};
+        r = {struct('s1', 1, 's2', 2, 'd1', 1, 'd2', 2),...
+             struct('s1', 1, 'd1', 1),...
+             struct('s1', 1, 's2', 2, 's3', 3, 'd1', 1, 'd2', 2)};
+    end
```

Comment:
I think it's worth adding a test case where the structs have different values for the same field.

Reply:
It will raise an error. I did try that yesterday but then I removed it. The approach used to combine structs is the first one [here](https://www.mathworks.com/matlabcentral/answers/96973-how-can-i-concatenate-or-merge-two-structures). The underlying assumption of the approach is that there will be no duplication of field names. But if that is a possibility then we need to define a policy of how to combine. So, that is why I left it simple.